### PR TITLE
[JENKINS-36592] Prevent NPE in BlockedBecauseOfBuildInProgress#getDescription() 

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It may happen due to the race condition.

1) AbstractProject#getCauseOfBlockage() checks for non-concurrent running build in order to generate a cause of blockage
2) If the getCauseOfBlockage() happens in the context not synced with queue, after isLogUpdated() && !isConcurrentBuild() there may be a context switch. 
3) Build may complete and get deleted before we actually call getLastBuild() in the code
4) If it happens, BlockedBecauseOfBuildInProgress may get null in the constructor
5) Then, BlockedBecauseOfBuildInProgress#getDescription() will start throwing NPEs

https://issues.jenkins-ci.org/browse/JENKINS-36592

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/model/AbstractProject.java [hunk 1]
```diff
@@ -1146,7 +1146,15 @@ public String getShortDescription() {
...
         // Block builds until they are done with post-production
         if (isLogUpdated() && !isConcurrentBuild()) {
-            return new BlockedBecauseOfBuildInProgress(getLastBuild());
+            final R lastBuild = getLastBuild();
+            if (lastBuild != null) {
+                return new BlockedBecauseOfBuildInProgress(lastBuild);
+            } else {
+                // The build has been lost during the execution
+                // It may happen in the case of the quick deletion of build after isLogUpdated() call 
+                // or API implemetation glisthes in the plugins. Anyway, we should let the code go then
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/AbstractProject.java]: "glitches"

File: core/src/main/java/hudson/model/AbstractProject.java [hunk 2]
```diff
@@ -1146,7 +1146,15 @@ public String getShortDescription() {
...
+                LOGGER.log(Level.FINEST, "Previous build instance has been during the non-concurrent cause creation. The build is not blocked anymore");
```
Comment (jglick) [inline core/src/main/java/hudson/model/AbstractProject.java]: Has been what?

Comment (jglick) [inline core/src/main/java/hudson/model/AbstractProject.java]: :ant: This does not seem like an appropriate use of `FINEST`. Use `FINE`.

**Comment by jglick** [comment]
:bee:

File: core/src/main/java/hudson/model/AbstractProject.java (hunk 2 above)
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/AbstractProject.java]: oops

**Comment by mafraba** [comment]
LGTM excepting some wording in the message/comments:

1. Some word missing in the message "Previous build instance has been during the non-concurrent cause creation".
2. Some words repeated in the comment "The build has been likely deleted after the isLogUpdated() the after isLogUpdated() call".

**Comment by oleg-nenashev** [comment]
@mafraba Sorry, my previous fast fix of the working just made the things worse. Should be fixed now

**Comment by oleg-nenashev** [comment]
@reviewbybees done from what I see

**Pull request merged**